src/GafferOSL/OSLObject.cpp

```diff
@@ -310,9 +316,23 @@ void OSLObject::hashProcessedObject( const ScenePath &path, const Gaffer::Contex
 	{
 		h.append( inPlug()->fullTransformHash( path ) );
 	}
-	if( useAttributesPlug()->getValue() )
+
+	if( gafferAttributes )
 	{
-		h.append( inPlug()->fullAttributesHash( path ) );
+		for( const auto &i : gafferAttributes->members() )
+		{
+			if( shadingEngine->needsAttribute( i.first ) )
+			{
+				if ( !i.second )
+				{
+					throw Exception( "Cannot compute hash from a CompoundObject will NULL data pointers!" );
```

Comment:
There's a typo in the message here, but I don't think we need this check anyway - everywhere else just assumes attributes have to be non-null.

Reply:
Oh, huh, I didn't want to change behaviour more than necessary, so I just copied this from Cortex ... I guess I should probably fix the typo here too: https://github.com/ImageEngine/cortex/blob/main/src/IECore/CompoundObject.cpp#L77

Reply:
Oh, doh.  Just saw the later comment about how it was never calling CompoundObject::hash in the first place.   Sorry for the confusion, though I suppose maybe I should still fix the Cortex typo